enterprise/server/workflow/config/config.go

```diff
@@ -37,6 +38,7 @@ type Action struct {
 	BazelWorkspaceDir string            `yaml:"bazel_workspace_dir"`
 	Env               map[string]string `yaml:"env"`
 	BazelCommands     []string          `yaml:"bazel_commands"`
+	MaxDuration       *time.Duration    `yaml:"max_duration"`
```

Comment:
nit: let's call this `timeout` to match Bazel's `--remote_timeout`, `--test_timeout`, `Action.timeout` etc.